api_tests/nodes/test_views.py

```diff
@@ -364,7 +372,116 @@ def test_incorrect_filtering_field_not_logged_in(self):
+        )
+
+        res = self.app.get(url, expect_errors=True)
+        assert_equal(res.status_code, http.BAD_REQUEST)
+        assert_equal(
+            res.json['errors'][0]['detail'],
+            "Value 'foo' is not valid for a filter on type bool."
+        )
+
+    def test_filter_by_date_created_invalid_date(self):
+        url = '/{0}nodes/?filter[date_created]=foo'.format(
+            API_BASE,
+        )
+
+        res = self.app.get(url, expect_errors=True)
+        assert_equal(res.status_code, http.BAD_REQUEST)
+        assert_equal(
+            res.json['errors'][0]['detail'],
+            "Value 'foo' is not valid for a filter on type date."
+        )
+
+    def test_filter_by_date_created_gt(self):
+        url = '/{0}nodes/?filter[date_created][gt]={1}'.format(
+            API_BASE,
+            datetime.datetime.isoformat(self.first_date)
+        )
+
+        res = self.app.get(url)
+        ids = [item['id'] for item in res.json['data']]
+        assert_not_in(self.project_one._id, ids)
+        assert_in(self.project_two._id, ids)
+        assert_in(self.project_three._id, ids)
+
+    def test_filter_by_date_created_gte(self):
+        url = '/{0}nodes/?filter[date_created][gte]={1}'.format(
+            API_BASE,
+            datetime.datetime.isoformat(self.first_date)
+        )
+
+        res = self.app.get(url)
+        ids = [item['id'] for item in res.json['data']]
+        assert_in(self.project_one._id, ids)
+        assert_in(self.project_two._id, ids)
+        assert_in(self.project_three._id, ids)
+
+    def test_filter_by_date_created_lt(self):
+        url = '/{0}nodes/?filter[date_created][lt]={1}'.format(
+            API_BASE,
+            datetime.datetime.isoformat(self.third_date)
+        )
+
+        res = self.app.get(url)
+        ids = [item['id'] for item in res.json['data']]
+        assert_in(self.project_one._id, ids)
+        assert_in(self.project_two._id, ids)
+        assert_not_in(self.project_three._id, ids)
+
+    def test_filter_by_date_created_lte(self):
+        url = '/{0}nodes/?filter[date_created][lte]={1}'.format(
+            API_BASE,
+            datetime.datetime.isoformat(self.third_date)
+        )
+
+        res = self.app.get(url)
+        ids = [item['id'] for item in res.json['data']]
+        assert_in(self.project_one._id, ids)
+        assert_in(self.project_two._id, ids)
+        assert_in(self.project_three._id, ids)
+
+    def test_filter_by_date_created_multiple_comparison(self):
+        url = '/{0}nodes/?filter[date_created][lt]={1}&filter[date_created][gt]={2}'.format(
+            API_BASE,
+            datetime.datetime.isoformat(self.third_date),
+            datetime.datetime.isoformat(self.first_date)
+        )
+        res = self.app.get(url)
+        ids = [item['id'] for item in res.json['data']]
+        assert_not_in(self.project_one._id, ids)
+        assert_in(self.project_two._id, ids)
+        assert_not_in(self.project_three._id, ids)
```

Comment:
I notice that your filter fields allow fuzzy matching, but I don't see any unit tests for that functionality. Did I miss some?

Since that's a subtle feature with edge cases, tests would be very helpful.